internal/vm/setup.go

```diff
@@ -76,6 +85,19 @@ _G["load_plugin_schema"] = function(plugin_schema_string)
   return plugin_name, nil
 end
 
+_G["unload_plugin_schema"] = function(plugin_name)
+  local base_error = "error unloading schema for plugin: "
+  if plugin_name == nil or plugin_name == "" then
+    return false, base_error .. "base_error"
+  end
+
+  if Plugins and Plugins.subschemas and Plugins.subschemas[plugin_name] then
```

Comment:
Just curious: in which circumstances this part of the check may fail?

`if Plugins and Plugins.subschemas`

Reply:
In our case `Plugins` is always instantiated at the global level so we can remove this useless check, but `Plugins.subschemas` may not be defined until a plugin schema is loaded; see `Schema.new_subschema` check [here](https://github.com/Kong/goks/blob/357cad4f6d81f99325de1bf2e20cdecd85083b38/lua-tree/share/lua/5.1/kong/db/schema/init.lua#L2362-L2364).

```suggestion
  if Plugins.subschemas and Plugins.subschemas[plugin_name] then
```